network/transport/grpc/connection_manager.go

```diff
@@ -414,3 +408,27 @@ func (s *grpcConnectionManager) constructMetadata() (metadata.MD, error) {
 	}
 	return md, nil
 }
+
+func (s *grpcConnectionManager) startConnecting(address string, connection Connection) {
```

Comment:
The name `startConnecting` is really confusing. Why start connecting on an existing and established connection?

Reply:
So that when the peer disconnects, the local node will try to reconnect for when it comes back.

Alternative would be dropping the outbound connector to the peer for which there is already an established (incoming) connection, but then the outbound address would be "lost" when the peer disconnects. Ofc. given the peer won't reconnect.

Thoughts?